File: projects/m1/016-next-day/solutions/Davide Di Giacomo/next_day.py
february = 2
months_30 = [4, 6, 9, 11]
months_31 = [1, 3, 5, 7, 8, 10, 12]
year_leap = 29

#insert a date
date = input("Enter a date in YYYY-MM-DD format: ")
year, month, day = map(int, date.split('-'))

#year
if month == 12 and day == 31:
    year+=1
    month = 1
    day = 1

#leap year or not
elif year % 400 == 0 and month == february and day == 28:
    day = year_leap
    
elif year % 100 == 0 and month == february and day == 28:
    month+=1
    day = 1
    
elif year % 4 == 0 and month == february and day == 28:
    day = year_leap
    
#28/30/31 day for month
elif month == february and day == 28 or day == year_leap:
    month +=1
    day = 1
    
elif month in months_30 and day == 30:
    month +=1
    day = 1
    
elif month in months_31 and day == 31:
    month +=1
    day = 1
    
else:
    day+=1

print (str(year) + "-" + str(month).zfill(2) + "-" + str(day).zfill(2))

File: projects/m1/016-next-day/solutions/stefandifede/next-day.js
let userYear = Math.floor(prompt("Insert the year"));
let userMonth = Math.floor(prompt("Insert the Month"));
let userDay = Math.floor(prompt("Insert the Day"));

//array of months with 30 days
const monthsOf30 = [4, 6, 9, 11]

//declaring a variable to check if inserted year is leap or not
let isLeap;
function isItLeap() {
    if (userYear % 400 == 0 || userYear % 100 != 0 && userYear % 4 == 0) {
           isLeap = true
    } else {
           isLeap = false
    }
}
isItLeap();


function monthOf28Days () {
    //checking if inserted day is a valid day, considering if the year is leap or not
    if ((userDay > 28 && isLeap == false) || (userDay > 29 && isLeap == true)) {
        alert("inserted date is invalid");
    /*if inserted day it's correct, add 1 day and change month if necessary*/
    } else if ((userDay + 1 > 28 && isLeap == false) || (userDay + 1 > 29 && isLeap == true)) {
        userDay = 1;
        userMonth = userMonth + 1;
        alert(userYear + "-" + userMonth + "-" + userDay);
    } else {alert(userYear + "-" + userMonth + "-" + Math.floor(userDay + 1))}
}

function monthsOf30Days () {
    if (userDay > 30) {
        alert("inserted date is invalid");
    } else if (userDay + 1 >= 30) {
        userDay = 1;
        userMonth = userMonth + 1;
        alert(userYear + "-" + userMonth + "-" + userDay);
    } else {alert(userYear + "-" + userMonth + "-" + Math.floor(userDay + 1))}
}

function monthsOf31Days () {
    if (userDay > 31) {
        alert("inserted date is invalid");
    /*if inserted day it's correct, add 1 day and change month and if necessary*/
    } else if (userDay + 1 >= 31) {
        userDay = 1;
        userMonth = userMonth + 1;
        alert(userYear + "-" + userMonth + "-" + userDay);
    } else {alert(userYear + "-" + userMonth + "-" + Math.floor(userDay + 1))}
}

function changeYear () {
    if (userDay > 31) {
        alert("inserted date is invalid");
    } else if (userDay + 1 > 31) {
        userDay = 1;
        userMonth = 1;
        userYear = userYear + 1;
        alert(userYear + "-" + userMonth + "-" + userDay);
    } else {alert(userYear + "-" + userMonth + "-" + Math.floor(userDay + 1))}
}

function changeMonth(){
    //if inserted month is february
    if (userMonth == 2) { 
        monthOf28Days()
    /*if inserted month is a month with 30 days*/ 
    } else if (monthsOf30.includes(userMonth)) {
        monthsOf30Days()
    /* if inserted month is not a month with 30 days (excluding December and February)*/
    } else if (monthsOf30.includes(userMonth) == false && userMonth != 12 && userMonth != 2) {
        monthsOf31Days()
    /*if inserted month is december*/
    } else if (userMonth == 12) {
        changeYear()
    }
}

changeMonth()